calculator/calculator.cpp

```diff
@@ -0,0 +1,202 @@
+    connect(ui->btnChangeSignal, SIGNAL(released()), this, SLOT(ChangeNumberSign()));
+
+    connect(ui->btnErase, SIGNAL(released()), this, SLOT(ClearSign()));
+
+    connect(ui->btnAddMemory, SIGNAL(released()), this, SLOT(AddToMemory()));
+    connect(ui->btnRemoveMemory, SIGNAL(released()), this, SLOT(RemoveFromMemory()));
+    connect(ui->btnRetrieveMemory, SIGNAL(released()), this, SLOT(DisplayMemory()));
+
+    connect(ui->btnBackspace, SIGNAL(released()), this, SLOT(Backspace()));
+
+}
+
+// Deconstructor
+Calculator::~Calculator()
+{
+    delete ui;
+}
+
+void Calculator::NumPressed()
+{
+    if (displayFlag)
+    {
+        ui->display->clear();
+        displayFlag = false;
+    }
+
+    QPushButton *btn = (QPushButton *)sender();
+    QString btnValue = btn->text();
+    QString displayValue = ui->display->text();
+
+    if ((displayValue.toDouble() == 0) || (displayValue.toDouble() == 0.0))
+    {
+        ui->display->setText(btnValue);
+    } else
+    {
+        QString newValue = displayValue + btnValue;
+        double doubleNewValue = newValue.toDouble();
+        ui->display->setText(QString::number(doubleNewValue, 'g', 16));
+    }
+}
+
+void Calculator::Backspace()
+{
+    QString currentValue = ui->display->text();
+    currentValue.chop(1); // remove last
+    if (currentValue.length() == 0)
+    {
+        ui->display->setText("0");
+    } else
+    {
+        ui->display->setText(currentValue);
+    }
+}
+
+void Calculator::MathButtonPressed()
+{
+    divTrigger = false;
+    multiTrigger = false;
+    addTrigger = false;
+    subTrigger = false;
+
+    QString currentValue = ui->display->text();
+    calcValue = currentValue.toDouble();
+
+    QPushButton *btn = (QPushButton *)sender();
+    QString btnValue = btn->text();
+
+    if (QString::compare(btnValue, "/", Qt::CaseInsensitive) == 0)
+    {
+        divTrigger = true;
+    }
+    else if (QString::compare(btnValue, "*", Qt::CaseInsensitive) == 0)
+    {
+        multiTrigger = true;
+    }
+    else if (QString::compare(btnValue, "+", Qt::CaseInsensitive) == 0)
+    {
+        addTrigger = true;
+    }
+    else
+    {
+        subTrigger = true;
+    }
+
+    // After an operation, we can clear our display (new number)
+    ui->display->setText(QString::number(calcValue) + " " + btnValue);
+}
+
+void Calculator::EqualButton()
+{
+    double result = 0.0;
+
+    QString currentvalue = ui->display->text();
+    QPushButton *btn = (QPushButton *)sender();
+    QString btnValue = btn->text();
+    double doubleCurrentValue = currentvalue.toDouble();
+
+    if (divTrigger || multiTrigger || addTrigger || subTrigger)
+    {
+        if (divTrigger)
+        {
+            result = calcValue / doubleCurrentValue;
+        } else if (multiTrigger)
+        {
+            result = doubleCurrentValue * calcValue;
+        } else if (addTrigger)
+        {
+            result = doubleCurrentValue + calcValue;
+        } else if (subTrigger)
```

Comment:
This else if is not consistent with last else on MathButtonPressed()

Reply:
I didn't understand. The identation?